baseband/vdif/base.py

```diff
@@ -462,6 +491,14 @@ def __init__(self, raw, nthread=1, sample_rate=None, header=None,
             if sample_rate is not None:
                 kwargs['sample_rate'] = sample_rate
             header = VDIFHeader.fromvalues(**kwargs)
+
+        # If header was passed but not sample_rate, extract sample_rate.
+        if sample_rate is None:
+            try:
+                sample_rate = header.sample_rate
+            except AttributeError:
+                pass  # super below will scan file to get sample rate.
```

Comment:
Why the `pass`? This is for the writer, so there cannot be a scan of the file. I think one should just raise an exception here.